File: src/types/depositWithdrawalRecord.ts
import type { CsvValidationError } from "./csvRecord";

export interface DepositWithdrawalPreviewRow {
  row: number;
  transaction_flag: string;
  transaction_category: string;
  amount: string;
  description: string;
  bank_name: string;
  branch_name: string;
  summary_content: string;
  edi: string;
}

export interface DepositWithdrawalReadResult {
  preview: DepositWithdrawalPreviewRow[];
  total_count: number;
  errors: CsvValidationError[];
}

File: src-tauri/src/domain/deposit_withdrawal_record.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DepositWithdrawalRecord {
    pub transaction_flag: String,
    pub transaction_category: String,
    pub amount: u64,
    pub description: String,
    pub bank_name: String,
    pub branch_name: String,
    pub summary_content: String,
    pub edi: String,
}
